### DISCUSSION TITLE
Conda stdpopsim run FutureWarnings

### DISCUSSION THREAD

**Discussion by raywray** [popsim-consortium/stdpopsim#1543] [Q&A]
Hi,

It looks like this issue is perhaps related to #1519 . I have followed the installation instructions on the github.io for miniconda. I am using Python 3.12.1. It seems that the installation was successful, but anytime I try to run an stdpopsim command (such as `stdpopsim --version` or `python -m stdpopsim`) I get the following error: 

```
/site-packages/stdpopsim/catalog/HomSap/demographic_models.py:158: FutureWarning: Calling int on a single element Series is deprecated and will raise a TypeError in the future. Use int(ser.iloc[0]) instead
  time=int(extended_GF.time.head(1) - 1), rate=0
/home/exouser/miniconda3/envs/genomic_speciation_env/lib/python3.12/site-packages/stdpopsim/catalog/HomSap/demographic_models.py:161: FutureWarning: Calling int on a single element Series is deprecated and will raise a TypeError in the future. Use int(ser.iloc[0]) instead
  time=int(extended_GF.time.tail(1) + 1), rate=0
```
I know that in the answer for #1519 this was deemed an upstream problem in tskit and a fix would come out shortly, but that was some time ago and I wonder if I've missed something, as there have been several releases of both stdpopsim and tskit since then. 

For now, I will try the solution of installing jsonschema==4.18 before installing stdpopsim, but would like to know if there is a long-term solution.

**Comment by petrelharp** [comment]
Thanks for the report. Just to clarify, @raywray - you say you've got an error, but `Error`s in python stop execution, while this is a `Warning` (which doesn't). Warnings like this can be safely ignored (as long as we understand what's causing them, which we do). Does the code actually run, or is there an Error happening elsewhere?

**Reply by petrelharp** [reply]
Hm, also - the issue you reference was fixed in tskit 0.5.6, released last October: https://github.com/tskit-dev/tskit/issues/2840#issuecomment-1755784251

**Reply by raywray** [reply]
I apologize, the output above seems just to be a warning. The code runs, I posted this here in anticipation of a conflict, as the warning states that there could be a `TypeError` in the future. When I run `stdpopsim` from the command line, I do, however, get the following `Errors`, which also don't stop the run:

```
--- Logging error ---
Traceback (most recent call last):
  File "/home/exouser/miniconda3/envs/genomic_speciation_env/lib/python3.12/logging/__init__.py", line 1160, in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
  File "/home/exouser/miniconda3/envs/genomic_speciation_env/lib/python3.12/logging/__init__.py", line 999, in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
  File "/home/exouser/miniconda3/envs/genomic_speciation_env/lib/python3.12/site-packages/stdpopsim/cli.py", line 60, in format
    r"Warning:\s*(.*?)\s*warnings.warn\(", record.args[0], re.DOTALL

IndexError: tuple index out of range
Call stack:
  File "/home/exouser/miniconda3/envs/genomic_speciation_env/bin/stdpopsim", line 10, in <module>
    sys.exit(stdpopsim_main())
  File "/home/exouser/miniconda3/envs/genomic_speciation_env/lib/python3.12/site-packages/stdpopsim/cli.py", line 1136, in stdpopsim_main
    run(args)
  File "/home/exouser/miniconda3/envs/genomic_speciation_env/lib/python3.12/site-packages/stdpopsim/cli.py", line 1127, in run
    args.runner(args)
  File "/home/exouser/miniconda3/envs/genomic_speciation_env/lib/python3.12/site-packages/stdpopsim/cli.py", line 856, in run_simulation
    ts = engine.simulate(**kwargs)
  File "/home/exouser/miniconda3/envs/genomic_speciation_env/lib/python3.12/site-packages/stdpopsim/slim_engine.py", line 1523, in simulate
    warnings.warn(
  File "/home/exouser/miniconda3/envs/genomic_speciation_env/lib/python3.12/warnings.py", line 109, in _showwarnmsg
    sw(msg.message, msg.category, msg.filename, msg.lineno,
```
My tskit version is 0.5.6, which should have fixed the issue according to your above comment

**Reply by petrelharp** [reply]
Hm, interesting - so, these errors don't stop execution? Can you post what command you ran to make that happen?

**Reply by jeromekelleher** [reply]
It looks like this is an error - I think what's happening is that @raywray is hitting a case where the SLiM engine is trying to emit a warning, but the message isn't properly formatted. There's some regex magic happening above that I don't understand, which may be the cause.

**Reply by raywray** [reply]
@jeromekelleher @petrelharp here is the command I used (from the website tutorial): `stdpopsim -e slim --slim-scaling-factor 10 HomSap -c chr22 --left 10000000 --right 20000000 -o foo1.ts -d Africa_1T12 AFR:100`

The run seems to work fine and successfully produces a `.ts` file.

**Reply by petrelharp** [reply]
Confirmed that this happens on my system as well. I'll close this discussion and open a bug report.

Thanks!!!

**Comment by petrelharp** [comment]
This seems to be errors that affect logging, but (probably) not the simulation output - moving further discussion to #1545.

Thanks!
